sopel/tools/target.py

```diff
@@ -193,13 +213,13 @@ def is_oper(self, nick):
             sensibly if only ``+v`` (voice) and ``+o`` (op) modes exist.
 
         """
-        return self.privileges.get(Identifier(nick), 0) & privileges.OPER
+        nick_id = self.make_identifier(nick)
+        return bool(self.privileges.get(nick_id, 0) & privileges.OPER)
```

Comment:
OK, this (and the numerous similar changes to other methods below) made me think: Why not just make `Channel.privileges` a `SopelIdentifierMemory`?

Reply:
Hm, I need to check that.

Reply:
Yes, this can be done, at a cost: we need to move `SopelIdentifierMemory` from `sopel.tools` to `sopel.tools.memory`, to prevent circular import.

If you want, I will be happy to do that in that PR, or into a future PR. Whatever you prefer!

Reply:
And this lead me to #2236 and #2237, the latter using `SopelIdentifierMemory` in `Channel`.